editor/post-title/index.js

```diff
@@ -35,17 +34,27 @@ class PostTitle extends Component {
 		this.onSelect = this.onSelect.bind( this );
 		this.onUnselect = this.onUnselect.bind( this );
 		this.onSelectionChange = this.onSelectionChange.bind( this );
+		this.onOuterBlur = this.onOuterBlur.bind( this );
+		this.onOuterFocus = this.onOuterFocus.bind( this );
+		this.setFocused = this.setFocused.bind( this );
+		this.focusText = this.focusText.bind( this );
+
 		this.state = {
 			isSelected: false,
+			hasFocusWithin: false,
 		};
+		this.blurTimer = null;
 	}
 
 	componentDidMount() {
 		document.addEventListener( 'selectionchange', this.onSelectionChange );
+		// eslint-disable-next-line react/no-find-dom-node
```

Comment:
The lint rule is correct here: `findDOMNode` should be discouraged, and this specific behavior can be recreated with a [`ref`](https://reactjs.org/docs/refs-and-the-dom.html). More ideally, we'd not rely on the DOM to determine how to assign focus state (the other way around being the preference, state -> DOM).

Also worth noting that this will always cause a re-render immediately after the first render, even if state is not changing (`setState` will always cause render, even if the `hasFocusWithin` is and stays `false`).

When would we expect this element to be the active element after a mount?

Reply:
Yeah, I'm new to React so I wasn't sure if this would ever be true. I guess it would be true if the input that appeared inside it had autofocus, but maybe not exactly on mounting. I'm happy to set it to false on startup and have nothing here.